auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2021-2024 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models, tools
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id"
+    )
+    server_action_id = fields.Many2one(
+        "ir.actions.server",
+        "Server Action",
+    )
+    log_selected_fields_only = fields.Boolean(
+        help="Log only the selected fields, to save space avoid large DB data.",
+    )
+
+    @api.model
+    @tools.ormcache("model_name")
+    def _get_field_names_of_rule(self, model_name):
+        """Memory-cached list of fields per rule"""
+        rule = (
+            self.env["auditlog.rule"]
+            .sudo()
+            .search([("model_id.model", "=", model_name)], limit=1)
+        )
+        if rule.auditlog_line_access_rule_ids:
+            return rule.mapped("auditlog_line_access_rule_ids.field_ids.name")
+        return []
+
+    @api.model
+    @tools.ormcache("model_name")
+    def _get_log_selected_fields_only(self, model_name):
+        """Memory-cached translation of model to rule"""
```

Comment:
@hbrunn I've noticed that as soon as we create an auditlog rule and subscribe, and then trigger a field write, nothing really happens, i.e., no auditlog.log(and line) records were created.

For this to work,  I had to unsubscribe, uncheck the log_selected_fields_only,  save form,  check the log_selected_fields_only,  save form,  and then subscribe again. That's the only way I could actually trigger account.log(and line) records to be created. Any thoughts on that?

Reply:
do a registry reload as in https://github.com/OCA/server-tools/blob/16.0/auditlog/models/rule.py#L249 below in write instead of just clearing the caches